### DISCUSSION TITLE
Is there a way to deserialize the `DMMF`?

### DISCUSSION THREAD

**Discussion by ndaba1** [prisma/prisma#17172] [Q&A] [answered]
I want to parse a .prisma file, modify the `dmmf` then deserialize it, preferably into a formatted string that can be written to a new file. I've been able to work out parsing the file and modifying the `dmmf` but I can't seem to find a way to convert it into a string.
Consider the following:
```js
import { getDMMF } from "@prisma/sdk"

const dmmf = await getDMMF({ datamodelPath: "somePath" })
// perform some modification
dmmf.datamodel.models = dmmf.datamodel.models.filter(m => m.name !== "test")

// missing implementation..... something like:
deserializeDMMF(dmmf) // TODO?
```
Is this something that can be achieved? If so, how?

**Comment by nurul3101** [comment, marked as answer]
Hello @ndaba1 👋 

For reference, DMMF is an internal API and can have breaking changes in minor updates.

That being said, you might want to have a look at [prisma-ast](https://github.com/MrLeebo/prisma-ast) and [prisma-schema-parser](https://github.com/loancrate/prisma-schema-parser) for looking into deserializing. 

Prisma AST has helper functions that return strings for a schema AST:

<img width="901" alt="Screenshot 2023-01-09 at 8 55 30 PM" src="https://user-images.githubusercontent.com/13049676/211344083-5bfb0836-ff6d-484f-8e3b-36a98387b077.png">

**Reply by ndaba1** [reply]
Thank you for this.... I had tried working with `prisma-schema-parser` but it didn't really meet my needs. But the `prisma-ast` pkg does. Thanks again!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
